### DISCUSSION TITLE
Lesson 6: Is a "0x" prefix needed for a $PRIVATE_KEY environment variable?

### DISCUSSION THREAD

**Discussion by TreapGoGo** [Cyfrin/foundry-full-course-cu#1112] [Q&A] [answered]
When I run 

```bash
forge script script/DeployRaffle.s.sol --rpc-url $SEPOLIA_RPC_URL --private-key $PRIVATE_KEY --broadcast
```

I got

```bash
[⠔] Compiling...
No files changed, compilation skipped
Traces:
  [639098] → new DevOpsTools@0x46bEFE6B3AecCcE19c77ae87A901aD622940D51e
    └─ ← 3192 bytes of code

  [21122157] → new DeployRaffle@0x5b73C5498c1E3b4dbA84de0F1833c4a029d90519
    └─ ← 105185 bytes of code

  [106375] DeployRaffle::run()
    ├─ [47531] → new <unknown>@0xC7f2Cf4845C6db0e1a1e91ED41Bcd0FcC1b0E141
    │   ├─ [0] VM::envUint("PRIVATE_KEY") [staticcall]
    │   │   └─ ← failed parsing $PRIVATE_KEY as type `uint256`: missing hex prefix ("0x") for hex string
    │   └─ ← 0 bytes of code
    └─ ← failed parsing $PRIVATE_KEY as type `uint256`: missing hex prefix ("0x") for hex string


Error: 
script failed: failed parsing $PRIVATE_KEY as type `uint256`: missing hex prefix ("0x") for hex string
```

So I went to my `.env` and added a "0x" prefix in front of my private key string, after which I run the same command as above and the contract was deployed successfully. I checked etherscan(link: [my contract](https://sepolia.etherscan.io/address/0x170f007dd52aA6937e18dB01C5423D7d88F3286A)) and found that **the creator of this contract matched my wallet**, which means adding a "0x" prefix to my private key seems not to change the content of my private key.

However, in Patrick's video, there is no "0x" prefix in `.env`, and the script can run. So I got confused: is this a update of Foundry?

**Comment by 0xusmanf** [comment, marked as answer]
> However, in Patrick's video, there is no "0x" prefix in .env, and the script can run. So I got confused: is this a update of Foundry?

Yes, there is an update to the Foundry, you can find this update [here](https://github.com/foundry-rs/foundry/pull/6269/files#diff-afeb9cd8026b22a646da7c752efc3cf61961fcf36c277f0c7d42d8df4c1cfa05).

**Reply by TreapGoGo** [reply]
Thanks for your reply ❤️
